crates/revm/src/db/states/bundle_state.rs

```diff
@@ -365,8 +365,28 @@ impl BundleState {
                                 .present_value = storage_slot.present_value;
                         }
                     }
-                    this.status = other_account.status;
                     this.info = other_account.info;
+                    // if this status was destroyed this means we know all the storages
+                    // and new status need to contains that "Destroyed" flag.
+                    this.status = match (
```

Comment:
does it make sense to declare `extend` fn on account status and put this logic there?

Reply:
It does make sense to do that. Moved this logic inside `AccountStatus::transition(..)`